src/variable.jl

```diff
@@ -203,7 +203,7 @@ Defines and return the variable in the data set `ds`
 copied from the variable `src`. The dimension name, attributes
 and data are copied from `src` as well as the variable name (unless provide by `name`).
 """
-function defVar(dest::AbstractDataset,varname::SymbolOrString,srcvar::AbstractVariable; kwargs...)
+function defVar(dest::AbstractDataset,varname::SymbolOrString,srcvar::Union{AbstractVariable, SubVariable}; kwargs...)
```

Comment:
I updated `::AbstractVariable` to `::Union{AbstractVariable, SubVariable}` the places I thought it made sense but I might have missed some.

Reply:
Isn't `SubVariable <: AbstractVariable` anyway ?

Reply:
yes, this is indeed the case

https://github.com/JuliaGeo/CommonDataModel.jl/blob/68f20501fe5327aa890d1ab71ba52972b140e004/src/types.jl#L158

Reply:
No, see the changes to types.jl

https://github.com/JuliaGeo/CommonDataModel.jl/pull/35/files#diff-525588e68b2421901be164965272940effa093de733a3fd36c4b9a4344b8c20cR55 

`SubVariable` is no longer a subtype of `AbstractVariable` but is now a subtype of `AbstractSubDiskArray`

```julia
struct SubVariable{T,N,P,I,L} <: DiskArrays.AbstractSubDiskArray{T,N,P,I,L}
```

This is done to use the DiskArray implementation of views as discussed in https://github.com/JuliaGeo/NCDatasets.jl/issues/274